cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerServiceTest.java

```diff
@@ -54,7 +67,11 @@ public static void finish() {
   @Test
   public void testSchedulesAcrossNamespace() throws Exception {
     AppFabricTestHelper.deployApplication(AppWithWorkflow.class);
+    ApplicationSpecification applicationSpecification = store.getApplication(appId);
+
     schedulerService.schedule(program, programType, ImmutableList.of(schedule1));
+    applicationSpecification = createNewSpecification(applicationSpecification, program, programType, schedule1);
```

Comment:
Since I introduced a check in the metastore everytime an operation on a schedule is done, to get the type of the schedule, it means that adding a schedule also means that the application specification has to be changed in the metastore, to contain the new schedule.
I don't know about other places in the code where we add a schedule to an existing app though? But I would expect that if there is, the code doing that would already modify the application specification to reflect the change.